shapely/geometry/base.py

```diff
@@ -105,6 +105,10 @@ def __new__(self):
         )
         return shapely.from_wkt("GEOMETRYCOLLECTION EMPTY")
 
+    @property
+    def geoms(self):
+        return GeometrySequence(self)
```

Comment:
@cookiedan42 here's what I suggest to solve the recursion issue. Do the following for the base class:

```suggestion
        raise NotImplementedError
```

And then for single-part `Point`, etc, classes, do this:

```python
@property
def geoms(self):
    return [self]
```

`GeometrySequence` is nothing more than a Python sequence protocol backed by a GEOS multi-part geometry, but as you notice, introduces some complication.